workers/api-server/src/http/wallet.rs

```diff
@@ -79,6 +80,50 @@ async fn append_task_and_await(
     Ok(task_id)
 }
 
+/// Create an order (potentially in a matching pool) and wait for it to be
+/// enqueued
+pub async fn create_order(
+    order: ApiOrder,
+    statement_sig: Vec<u8>,
+    wallet_id: WalletIdentifier,
+    state: &State,
+    matching_pool: Option<MatchingPoolName>,
+) -> Result<CreateOrderResponse, ApiServerError> {
+    let id = order.id;
+
+    // Lookup the wallet in the global state
+    let old_wallet = find_wallet_for_update(wallet_id, state).await?;
+    let mut new_wallet = old_wallet.clone();
+    let new_order: Order = order.into();
+
+    // Check that the timestamp is not too old, then add to the wallet
+    new_wallet.add_order(id, new_order.clone()).map_err(bad_request)?;
+    new_wallet.reblind_wallet();
+
+    let task = match matching_pool {
```

Comment:
feel like it'd be cleaner to just create a more direct constructor that accepts an option; like `new_with_with_maybe_pool`